template.yaml

```diff
@@ -383,3 +386,132 @@ Resources:
+          Id: ExternalUpdatesQueueTarget
+          DeadLetterConfig:
+            Arn: !GetAtt ExternalUpdatesDLQ.Arn
+
+  ExternalUpdatesQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref ExternalUpdatesQueue
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: "events.amazonaws.com"
+            Action: 'sqs:SendMessage'
+            Resource: !GetAtt ExternalUpdatesQueue.Arn
+            Condition:
+              ArnEquals:
+                aws:SourceArn: !GetAtt ExternalUpdateEventsRule.Arn
+
+  ConsumeMessagesFromExternalEventsQueueManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: 'Managed IAM policy to consume messages from external events queue'
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - sqs:ReceiveMessage
+              - sqs:DeleteMessage
+              - sqs:GetQueueAttributes
+            Resource: !GetAtt ExternalUpdatesQueue.Arn
+
+  ReadDataCiteCredentialsManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: 'Managed IAM policy to read secrets required to authenticate to datacite on behalf of customers'
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+            Resource:
+              Fn::Sub: arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${CustomerSecretsSecretName}*
+
+  ReadEventBucketManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: 'Managed IAM policy to consume messages from external events queue'
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - s3:GetObject
+            Resource: !Sub "arn:aws:s3:::${EventsBucketsName}/*"
+
+  ExternalUpdatesEventHandler:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: datacite-delete-draft-doi-handler
+      Handler: no.unit.nva.datacite.handlers.resource.ExternalUpdatesEventHandler::handleRequest
+      Timeout: 900
+      Policies:
+        - !Ref ReadDataCiteCredentialsManagedPolicy
+        - !Ref ConsumeMessagesFromExternalEventsQueueManagedPolicy
+        - !Ref ReadEventBucketManagedPolicy
+      Environment:
+        Variables:
+          EVENTS_BUCKET_NAME: !Ref EventsBucketsName
+          CUSTOMER_SECRETS_SECRET_NAME: !Ref CustomerSecretsSecretName
+          CUSTOMER_SECRETS_SECRET_KEY: !Ref CustomerSecretsSecretKey
+          DATACITE_REST_HOST: !Ref DataCiteRestHost
+          DATACITE_MDS_HOST: !Ref DataCiteMdsHost
+          DOI_HOST: !Ref DoiHost
+      Events:
+        ExternalUpdatesQueueEvent:
+          Type: SQS
+          Properties:
+            Queue: !GetAtt ExternalUpdatesQueue.Arn
```

Comment:
Should we add property -> `BatchSize: 1`
For now handler is set to handler multiple sqs messages per single invocation:

```
Optional.ofNullable(sqsEvent.getRecords()).stream()
            .flatMap(List::stream)
            .map(ExternalUpdatesEventHandler::parseEventReference)
            .filter(Objects::nonNull)
            .forEach(this::processPayload);
```

With current implementation when processing of first message will fail, all the messages in SQSEvent will also fail and will not be processed.

Reply:
Will set BatchSize to 1.